src/arithmetic/arithmetic_expression.c

```diff
@@ -389,10 +432,12 @@ static AR_EXP_Result _AR_EXP_Evaluate(AR_ExpNode *root, const Record r, SIValue
 SIValue AR_EXP_Evaluate(AR_ExpNode *root, const Record r) {
 	SIValue result;
 	AR_EXP_Result res = _AR_EXP_Evaluate(root, r, &result);
-	if(res != EVAL_OK) {
+	if(res == EVAL_ERR) {
 		QueryCtx_RaiseRuntimeException();  // Raise an exception if we're in a run-time context.
 		return SI_NullVal(); // Otherwise return NULL; the query-level error will be emitted after cleanup.
 	}
+	// Found a parameter in the expression tree, try to reduce the tree.
```

Comment:
Please rephrase
`// At least one param node was encountered during evaluation, tree should be param node free, try reducing the tree.`